salt/modules/openscap.py

```diff
@@ -55,6 +58,132 @@ def error(self, message, *args, **kwargs):
+    in the master's minion files cachedir
+    (defaults to ``/var/cache/salt/master/minions/minion-id/files``)
+
+    It needs ``file_recv`` set to ``True`` in the master configuration file.
+
+    xccdffile
+        the path to the xccdf file to evaluate
+
+    ovalfiles
+        additional oval definition files
+
+    profile
+        the name of Profile to be evaluated
+
+    rule
+        the name of a single rule to be evaluated
+
+    oval_results
+        save OVAL results as well (True or False)
+
+    results
+        write XCCDF Results into given file
+
+    report
+        write HTML report into given file
+
+    fetch_remote_resources
+        download remote content referenced by XCCDF (True or False)
+
+    tailoring_file
+        use given XCCDF Tailoring file
+
+    tailoring_id
+        use given DS component as XCCDF Tailoring file
+
+    remediate
+        automatically execute XCCDF fix elements for failed rules.
+        Use of this option is always at your own risk. (True or False)
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*'  openscap.xccdf_eval /usr/share/openscap/scap-yast2sec-xccdf.xml profile=Default
+
+    """
+    success = True
+    error = None
+    upload_dir = None
+    returncode = None
+    if not ovalfiles:
+        ovalfiles = []
+
+    cmd_opts = ["oscap", "xccdf", "eval"]
+    if oval_results:
+        cmd_opts.append("--oval-results")
+    if results:
+        cmd_opts.append("--results")
+        cmd_opts.append(results)
+    if report:
+        cmd_opts.append("--report")
+        cmd_opts.append(report)
+    if profile:
+        cmd_opts.append("--profile")
+        cmd_opts.append(profile)
+    if rule:
+        cmd_opts.append("--rule")
+        cmd_opts.append(rule)
+    if tailoring_file:
+        cmd_opts.append("--tailoring-file")
+        cmd_opts.append(tailoring_file)
+    if tailoring_id:
+        cmd_opts.append("--tailoring-id")
+        cmd_opts.append(tailoring_id)
+    if fetch_remote_resources:
+        cmd_opts.append("--fetch-remote-resources")
+    if remediate:
+        cmd_opts.append("--remediate")
+    cmd_opts.append(xccdffile)
+    cmd_opts.extend(ovalfiles)
+
+    if not os.path.exists(xccdffile):
+        success = False
+        error = "XCCDF File '{}' does not exist".format(xccdffile)
+    for ofile in ovalfiles:
+        if success and not os.path.exists(ofile):
+            success = False
+            error = "Oval File '{}' does not exist".format(ofile)
+
+    if success:
+        tempdir = tempfile.mkdtemp()
+        proc = Popen(cmd_opts, stdout=PIPE, stderr=PIPE, cwd=tempdir)
+        (stdoutdata, error) = proc.communicate()
+        success = _OSCAP_EXIT_CODES_MAP.get(proc.returncode, False)
+        if proc.returncode < 0:
+            error += "\nKilled by signal {}\n".format(proc.returncode).encode("ascii")
+        returncode = proc.returncode
+        if success:
+            __salt__["cp.push_dir"](tempdir)
```

Comment:
I think it would make sense to take `cp.push_dir`'s return value into account. As far as I can tell from skimming its code, it will indicate if pushing the files was successful. If it wasn't, I would report that to the caller.

Reply:
Added your suggestion here @agraul